lib/ecto/query/planner.ex

```diff
@@ -1051,7 +1051,7 @@ defmodule Ecto.Query.Planner do
           # We don't want to use normalize_subquery_select because we are
           # going to prepare the whole query ourselves next.
           {_, _, inner_query} = rewrite_subquery_select_expr(inner_query, true)
-          {inner_query, counter} = traverse_exprs(inner_query, :all, counter, fun)
+          {inner_query, counter} = traverse_exprs(inner_query, opts[:operation], counter, fun)
```

Comment:
since this is a map, this should work, right?

```suggestion
          {inner_query, counter} = traverse_exprs(inner_query, opts.operation, counter, fun)
```

Reply:
Yes, you're right, I had forgotten it's always a map. Resolved in https://github.com/elixir-ecto/ecto/pull/4220/commits/b1667db38f24b39090b36ae5373589de5b273896